### PR TITLE
Opencmw sink

### PR DESCRIPTION
**Pull request by sfalamic**
First practices with GNURadio, implementation of GNURadio Sink

### CODE REVIEW

File: grc/pulsed_power_daq_power_calc_mul_ph_ff.block.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+id: pulsed_power_daq_power_calc_mul_ph_ff
```
Comment (alexxcons) [inline grc/pulsed_power_daq_power_calc_mul_ph_ff.block.yml:1]: This block looks like it would not be related to the opencmw sink ... possibly should be a separate pull request ?

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +71,21 @@ endif(APPLE)
...
+FetchContent_Declare(
+        opencmw-cpp
+        GIT_REPOSITORY https://github.com/fair-acc/opencmw-cpp.git
+        GIT_TAG main # todo: use proper release once available
+)
+
+set(ENABLE_TESTING OFF)
+FetchContent_MakeAvailable(opencmw-cpp)
+target_link_libraries(gnuradio-pulsed_power_daq majordomo)
+
```
[the remark this answers is not in the document]
Comment (wirew0rm) [reply lib/CMakeLists.txt:88]: yes, we are discussing a similar concept to decouple the gnuradio part from the opencmw part. Additionally this is not meant to be merged into the main branch at the moment.

File: grc/pulsed_power_daq_opencmw_sink.block.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#     * dtype (e.g. int, float, complex, byte, short, xxx_vector, ...)
+#     * default
+parameters:
+- id: workers_count
+  label: Number of workers
+  dtype: int
+  default: 1
+#- id: ...
+#  label: ...
+#  dtype: ...
```
[the remark this answers is not in the document]
Comment (wirew0rm) [reply grc/pulsed_power_daq_opencmw_sink.block.yml:22]: yes, github also has the feature, accessible with the dropdown on the `Create PR` button when creating a pull request and some time ago they also added the possibilitiy to change to draft afterwards (hidden in the links in the right column with the labels and other PR metadata)

File: grc/pulsed_power_daq_power_calc_mul_ph_ff.block.yml (hunk 1 above)
Comment (wirew0rm) [reply grc/pulsed_power_daq_power_calc_mul_ph_ff.block.yml:1]: This is because this was based on the [pulsed_power_daq-dev](https://github.com/fair-acc/gr-digitizers/tree/pulsed_power_daq-dev) branch but i set the `pulsed_power_daq-main` as the base branch... I'll try to do some cleanup there.

File: lib/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexxcons) [reply lib/CMakeLists.txt:88]: Alright .. in case you decide to use the concept .. [here](https://gitlab.com/al.schwinn/DigitizerClass2/-/blob/master/src/DigitizerClass2/Common/Sink.cpp#L217) the related call on the fesa side. An object is passed as `userdata` and later statically cast back [inside the callback method](https://gitlab.com/al.schwinn/DigitizerClass2/-/blob/master/src/DigitizerClass2/Common/Sink.cpp#L52).

**Comment by wirew0rm** [comment]
We decided to move the contents of this PR into the corresponding project repository. See [fair-acc/pulsed-power-ml opencmw_sink_worker branch](https://github.com/fair-acc/pulsed-power-ml/tree/opencmw_sink_worker).
